### ISSUE TITLE
Panel site history

### ISSUE THREAD

**Issue by foxdoescode** [notion-enhancer/notion-enhancer#343]
**problem**
I embed google using panel sites to do research while taking notes and it would be useful to go back to the previous page.

**solution**
how will this feature appear/act?
![image](https://user-images.githubusercontent.com/75043185/101899161-5780b100-3b7b-11eb-8f15-4d0386e44564.png)

This would require extending the panel options to either include back and forward as even more special cases or make a general icon api that would allow each mod to implement its own icons.

For storing the history, it would just require an onload listener, a list of urls, and an index.

**alternatives**
one could use a browser and constantly switch between the browser and notion but that context switching is unnecessary.

**Comment by dragonwocky** [comment]
i think this would be at a level suitable for its own mod to have a full search engine/browser running inside notion, not a simple part of the current panel sites mod.

**Comment by foxdoescode** [comment]
Yeah that makes sense. I think it would still be valuable to have a more general icon api. 

Also I don't see why there has to be panel-site specific code in the general side panel code so maybe an api to add multiple panel pages.

**Comment by CloudHill** [comment]
i did try to implement a back and forward button, but since the iframe is cross-origin i couldn't access the current url of the iframe (`iframe.contentWindow.location.*` is blocked). there's also a weird thing going on where opening links in the iframe affects `window.history` so the history ends up being a mix of notion pages and the iframe's pages, and using notion's back button or calling `history.back()` alsos navigate the iframe. i did try getting the url through the history, but it doesn't seem to give access to the urls.

i admit having panel-sites code inside the side panel code is bad design, it's a quick n easy temp thing just to get it out for everyone to try it out. the buttons/icons api is also planned, but i'm working on other things atm.

all in all, i'm not sure if the way i'm implementing the side panel is the best, so if anyone would like to apply some optimizations or change some stuff up, feel free to do so.

**Comment by foxdoescode** [comment]
I was playing around a bit and have two ideas/solutions that are hacky but:

1. You can intercept subFrame requests to get the url. I just tested it out with this (which worked):
```js
mainWindow.webContents.session.webRequest.onCompleted((details, callback) => {
    if(details.resourceType === "subFrame") {
        console.log(details.url, details.referrer, details.resourceType);
    }
})
```
This would make the history of the sites and any other subframes interact with each other but one could track which site is the current one and add it to the correct stack.

2. You could try to enable webviews and then use a webview but I haven't tried it yet.

**Comment by dragonwocky** [comment]
@foxdoescode only thing with 1 is that it could still mess you up if you have any other embedded iframes in the page, or if you load another site in a different "tab" through panel sites.

**Comment by foxdoescode** [comment]
I mentioned that. I don't know it there's a solution for non-tab embedded frames but one can keep track of the current tab.
